### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by dignifiedquire**
tests are a bit flaky but they generally work

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        cb()
+      }
+    ], done)
+  })
+
+  // skipping, as otherwise CI will randomly break
+  it.skip('calls find peer on the connected node (using ipfs.io)', function (done) {
+    this.timeout(100000)
+    const routing = new DelegatedPeerRouting()
+    const id = 'Qmbut9Ywz9YEDrz8ySBSgWyJk41Uvm2QJPhwDJzJyGFsD6'
```
Comment (lgierth) [inline test/index.spec.js]: It's better to use one of the QmSoL bootstrap nodes here, currently we plan to run them for years to come

**Comment by VictorBjelkholm** [comment]
Tests are passing. Gonna retry a couple of times to check the flakiness. Once done, I'll rebase everything.

**Comment by VictorBjelkholm** [comment]
Tests now pass 99% of the time, unless we completely remove timeouts, it'll probably be like that as the tests call out into the network

image

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        ipfsd.stop(cb)
+      }
+    ], done)
+  })
+
+  it('calls find peer on the connected node (using ipfs.io)', function (done) {
+    this.timeout(1000 * 120)
+    const routing = new DelegatedPeerRouting()
+    // Solus Bootstrapper Node ID
+    const id = 'QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu'
```
Comment (lgierth) [inline test/index.spec.js]: This is saturn which is in Singapore (= shitty connectivity), better use `QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm` (uranus) or `QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64` (venus) which are in NYC.

Sorry I didn't think of this when you asked earlier.

(Pro tip: the first character after QmSoL is the first letter of the hostname, i.e. s=saturn, v=venus, u=uranus)

**Comment by diasdavid** [comment]
Merging this one in so that we can continue to iterate over separate PRs.

**Pull request merged**